### DISCUSSION TITLE
Led Effects, why only two?

### DISCUSSION THREAD

**Discussion by Morcegolas** [SmartHome-yourself/sonoff-tx-ultimate-for-esphome#63] [Q&A] [answered]
First of all thanks for making this possible. 
I want to ask why when selecting NeoPixel effects it only have two of them when in original firmware it had 3 or 4, I remember at least one more that was the laser one. Is it possible to have more effects?
Thanks.

**Comment by barth11** [comment, marked as answer]
I don't speak Spanish but this guy included many more light effects:
He's showing it in this part: https://youtu.be/58v8oqSQgXQ?si=_M_QXHQ3vUyeBEzr&t=1274
The definition is at 18:50.
Rainbow and Pulse are defined here: https://github.com/SmartHome-yourself/sonoff-tx-ultimate-for-esphome/blob/1723392ac7a1cf0e4ad540aa97585974668abbed/tx_ultimate.yaml#L359C5-L359C12
I would assume this it the only one place where it's needed to be adjusted for you to get more effects. Not sure how to do it in Substitution, if possible.
These should be possible when you expand the code:
https://esphome.io/components/light/#light-effects

**Reply by Morcegolas** [reply]
Many thanks, that was I was looking for.
